src/main/java/org/eclipse/yasson/internal/Marshaller.java

```diff
@@ -68,22 +68,50 @@ public Marshaller(JsonbContext jsonbContext) {
      *
      * @param object object to marshall
      * @param jsonGenerator generator to use
+     * @param close if generator should be closed
      */
-    public void marshall(Object object, JsonGenerator jsonGenerator) {
+    public void marshall(Object object, JsonGenerator jsonGenerator, boolean close) {
         try {
             serializeRoot(object, jsonGenerator);
         } catch (JsonbException e) {
             logger.severe(e.getMessage());
             throw e;
+        } catch (Exception e) {
+            logger.severe(e.getMessage());
+            throw new JsonbException(Messages.getMessage(MessageKeys.INTERNAL_ERROR, e.getMessage()), e);
         } finally {
             try {
-                jsonGenerator.close();
+                if (close) {
+                    jsonGenerator.close();
+                }
             } catch (JsonGenerationException jge) {
                 logger.severe(jge.getMessage());
             }
         }
     }
 
+    /**
+     * Marshals given object to provided Writer or OutputStream.
+     * Closes the generator on completion.
+     *
+     * @param object object to marshall
+     * @param jsonGenerator generator to use
+     */
+    public void marshall(Object object, JsonGenerator jsonGenerator) {
+        marshall(object,jsonGenerator,true);
+    }
+
+    /**
+     * Marshals given object to provided Writer or OutputStream.
+     * Leaves generator open for further interaction after completion.
+     *
+     * @param object object to marshall
+     * @param jsonGenerator generator to use
+     */
+    public void marshallWithoutClose(Object object, JsonGenerator jsonGenerator) {
```

Comment:
this method doesn't seem necessary

Reply:
This method is called from new JsonBinding impl methods. When JsonGenerator is passed to the runtime it is supposed to be managed externally.